packages/lesswrong/server/manualMigrations/migrationUtils.ts

```diff
@@ -65,13 +66,31 @@ export async function runMigration(name: string)
     started: new Date(),
   });
   
+  const db = getSqlClient();
+
+  // We can't assume that certain postgres functions exist because we may not have run the appropriate migration
+  // This wraapper runs the function and ignores if it's not defined yet
+  const safeRun = async (fn) => {
+    if(!db) return;
+
+    await db.any(`DO $$
+      BEGIN
+          PERFORM ${fn}();
+      EXCEPTION WHEN undefined_function THEN
+        -- Ignore if the function hasn't been defined yet; that just means migrations haven't caught up
+      END;
+    $$;`)
+  }
+
+  // TODO: do this atomically in a single transaction
   try {
+    await safeRun(`remove_lowercase_views`) // Remove any views before we change the underlying tables
```

Comment:
I could be misunderstanding, but I think this is in the wrong place.

All (or at least, most) changes to the Postgres schema should be done in automatic migrations, not in manual migrations, so this should probably be called somewhere in `packages/lesswrong/server/migrations/meta/umzug.ts`.

I don't think it hurts to have it here as well though.